### PR TITLE
Feat/early dns validation

### PR DESCRIPTION
**Pull request by puayhiang**
## Summary

Earlier checks for DNS TXT upon reaching FormSelect Page

## Changes

Shifted DNS Check from onClick to useEffect

## Issues

NIL

### CODE REVIEW

File: src/components/FormSelectionContainer/FormSelect/FormSelect.tsx [hunk 1]
```diff
@@ -18,27 +18,28 @@ export const FormSelect: FunctionComponent<FormSelectProps> = ({ id, form, onAdd
...
-  };
+  }, [config, form]);
 
   useEffect(() => {
-    if (isValidDns) {
-      onAddForm();
-    }
-  }, [isValidDns, onAddForm]);
+    dnsCheck();
+  }, [dnsCheck]);
```
Comment (simboonlong) [inline src/components/FormSelectionContainer/FormSelect/FormSelect.tsx:43]: oh think the intention was avoid making unnecessary calls to google dns, for as long as user did not initiate click

**Comment by puayhiang** [comment]
Closed based on discussions.
tl;dr: Running checks on all forms on load is a bit of an overkill, especially given that leeway (Pre-approvals) has to be given on slower or failing calls.
https://github.com/TradeTrust/document-creator-website/pull/228#discussion_r825542274
